topics/statistics/tutorials/RNN/tutorial.md

```diff
@@ -0,0 +1,365 @@
+and TF-IDF are suitable representations for when word order is not important. They are used in document classification problems like spam detection.
+
+## One hot encoding (OHE)
+
+OHE is a technique to convert categorical variables such as words into a vector. Suppose our vocabulary has 3 words: orange, apple, banana. 
+Each word for this vocabulary is represented by a vector of size 3. Orange is represented by a vector whose first element is 1 and other 
+elements are 0; Apple is represented by a vector whose second element is 1 and other elements are 0; And banana is represented by a 
+vector whose third element is 1 and other elements are 0. As you can see only one element in the vector is 1 and the rest are 0's. The same 
+concept applies if the size of the vocabulary is N.    
+
+![Alternative text](../../images/OHE.gif "One hot encoding (OHE) representation")
+
+The problem with OHE is that for very large vocabulary sizes (say, 100,000 words) it requires tremendous amount of storage. Also, it has no 
+concept of word similarity.   
+
+## Word2Vec
+
+In Word2Vec, each word is represented as an n dimensional vector (n being much smaller than vocabulary size), such that the words that have 
+similar meanings are closer to each other in the vector space, and words that don't have a similar meaning are farther apart. Words are 
+considered to have a similar meaning if they co-occur often in documents. There are 2 Word2Vec architectures, one that predicts the probability 
+of a word given the surrounding words (Continuous BOW), and one that given a word predicts the probability of the surrounding words (Continuous 
+skip-gram).
+
+In this tutorial, we find an n dimensional representation of the IMDB movie review words, not based on word meanings, but based on how they
+improve the sentiment classification task. The n dimensional representation is learned by the learning algorithm, simply by reducing the 
+cost function via backpropagation. 
+
+# Get data
+
+> ### {% icon hands_on %} Hands-on: Data upload
+>
+> - {% tool [Import some data](upload1) %}
+>    - Create a new history for this tutorial
+>    - Import the files from [Zenodo](https://zenodo.org/record/4477881#.YBrEQ3dKgvo) or from the shared data library
+>
+>    ```
+>    https://zenodo.org/record/4477881/files/X_test.tsv
+>    https://zenodo.org/record/4477881/files/X_train.tsv
+>    https://zenodo.org/record/4477881/files/y_test.tsv
+>    https://zenodo.org/record/4477881/files/y_train.tsv
+>    ```
+>    {% include snippets/import_via_link.md %}
+>    {% include snippets/import_from_data_library.md %}
+>
+>    - Rename the datasets as `X_test`, `X_train`, `y_test`, and `y_train` respectively
+>
+>    {% include snippets/rename_dataset.md %}
+>
+>    - Check that the datatype of all the four datasets is `tabular`. If not, change the dataset's datatype to tabular. 
+>
+>    {% include snippets/change_datatype.md datatype="datatypes" %}
```

Comment:
```suggestion
>    {% include snippets/change_datatype.md datatype="tabular" %}
```

that datatype var is used in the snippet, and should match your desired target datatype.

Reply:
Done